Py time (#669)
Codecademy/docs: content/python/concepts/time-module/terms/time-strftime/time-strftime.md

---
Title: 'time.strftime()'
Description: 'Returns a time as a string based on one or more formatting codes.'
Subjects:
  - 'Computer Science'
  - 'Data Science'
  - 'Web Development'
Tags:
  - 'Methods'
  - 'Constants'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
  - 'paths/front-end-engineer-career-path'
---

The `strftime()` function returns a time string based on one or more formatting codes and a time represented as a tuple or `stuct_time` format.

## Syntax

```pseudo
strftime(format_codes, time_value)
```

A combination of `format_codes` are used to format the returned string. The `time_value` is the time that will be formatted and can be represented as a [tuple](https://www.codecademy.com/resources/docs/python/tuples) or `struct_time` value. If no `time_value` is provided the function will return the current local time as formatted by the given expression.

A few common codes are:

- %a - The abbreviated day of the week (e.g. Mon)
- %d - The day of month as two decimals (08)

> - %Y - The calendar year represented as four decimals (2020)

Reviewer: The list of codes should be more complete. It should also definitely list the codes used in the examples.

---

> Title: 'time.strftime()'

Reviewer 2: Either way works here, I think, because you can introduce `strftime()` (or most modules?) in two ways:

```
from time import strftime
# "strftime()"

import time
# ".strftime()"

Description: 'Returns a time as a string based on one or more formatting codes.'
Subjects:
  - 'Computer Science'
  - 'Data Science'
  - 'Web Development'
Tags:
  - 'Methods'
  - 'Constants'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
  - 'paths/front-end-engineer-career-path'
---

Returns a time as a string based on one or more formatting codes.

## Syntax

```pseudo

> strftime(formatting_exp, time_value)

Reviewer 2: ```suggestion
strftime(format_codes, time_value)
```

```

The argument `formatting_exp` is the combination of formatting codes to be applied to the time input. The `time_value` is the time to be formatted. If no argument is provided the function will return the current local time as formatted by the given expression.

A few common codes are:

> - %a - Day name abbreviation (Mon)
> - %d - Day of month in two decimal format (28)
> - %Y - Year in four decimal format (2020)

Reviewer 2: ```suggestion
- %a - The abbreviated day of the week (e.g. Mon).
- %d - The day of the month as one or two decimals (1-31).
- %Y - The calendar year represented as four decimals (2020).
```

## Example

The `strftime()` function can be used to translate a time to display the associated date:

```py
import time

print(time.strftime('%x,time.ctime(1648501804)))
# Output: 03/28/22
```

## Codebyte Example

> The `strftime()` function can be used to return the current local time as the day and time formatted using a 12 hour clock format.

Reviewer 2: ```suggestion
The `strftime()` function can be used to return the current local time with the day and time formatted for a 12-hour clock:
```
